pkg/controller/logstorage/secrets/secret_controller.go

```diff
@@ -0,0 +1,619 @@
+// It provisions secrets and the trusted bundle into the requisite namespace(s) for other controllers to consume.
+type SecretSubController struct {
+	client        client.Client
+	scheme        *runtime.Scheme
+	status        status.StatusManager
+	clusterDomain string
+	multiTenant   bool
+}
+
+func Add(mgr manager.Manager, opts options.AddOptions) error {
+	if !opts.EnterpriseCRDExists {
+		return nil
+	}
+
+	// Create the reconciler
+	r := &SecretSubController{
+		client:        mgr.GetClient(),
+		scheme:        mgr.GetScheme(),
+		clusterDomain: opts.ClusterDomain,
+		multiTenant:   opts.MultiTenant,
+		status:        status.New(mgr.GetClient(), "log-storage-secrets", opts.KubernetesVersion),
+	}
+	r.status.Run(opts.ShutdownContext)
+
+	// Create a controller using the reconciler and register it with the manager to receive reconcile calls.
+	c, err := controller.New("log-storage-secrets-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Configure watches for operator.tigera.io APIs this controller cares about.
+	if err = c.Watch(&source.Kind{Type: &operatorv1.LogStorage{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch LogStorage resource: %w", err)
+	}
+	if err = utils.AddNetworkWatch(c); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch Network resource: %w", err)
+	}
+	if err = c.Watch(&source.Kind{Type: &operatorv1.ManagementCluster{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch ManagementCluster resource: %w", err)
+	}
+	if err = c.Watch(&source.Kind{Type: &operatorv1.ManagementClusterConnection{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch ManagementClusterConnection resource: %w", err)
+	}
+	if err = utils.AddTigeraStatusWatch(c, "log-storage-secrets"); err != nil {
+		return fmt.Errorf("logstorage-controller failed to watch logstorage Tigerastatus: %w", err)
+	}
+	if opts.MultiTenant {
+		if err = c.Watch(&source.Kind{Type: &operatorv1.Tenant{}}, &handler.EnqueueRequestForObject{}); err != nil {
+			return fmt.Errorf("log-storage-controller failed to watch Tenant resource: %w", err)
+		}
+	}
+
+	// The namespace(s) we need to monitor depend upon what tenancy mode we're running in.
+	installNS := render.ElasticsearchNamespace
+	truthNS := common.OperatorNamespace()
+	if opts.MultiTenant {
+		// For multi-tenant, the manager could be installed in any number of namespaces.
+		// So, we need to watch the resources we care about across all namespaces.
+		installNS = ""
+		truthNS = ""
+	}
+
+	// Watch all the elasticsearch user secrets in the operator namespace.
+	// TODO: In the future, we may want put this logic in the utils folder where the other watch logic is.
```

Comment:
This TODO pre-dates my changes. but I will probably do it in this PR anyway. Not sure why this wasn't done in the first place.